server/sonar-server/src/main/java/org/sonar/server/qualitygate/changeevent/QGChangeEventListener.java

```diff
@@ -20,7 +20,40 @@
 package org.sonar.server.qualitygate.changeevent;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import org.sonar.api.rules.RuleType;
 
 public interface QGChangeEventListener {
-  void onChanges(Trigger trigger, Collection<QGChangeEvent> changeEvents);
+  /**
+   * @deprecated use {{@link #onIssueChanges(QGChangeEvent, Set)}} instead
+   */
+  // TODO remove this method and turn default method #onIssueChanges(QGChangeEvent, Set) into an interface method when
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)